include/iterlib/RocksDBIterator.h

```diff
@@ -0,0 +1,65 @@
+//  Copyright (c) 2016, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+#pragma once
+
+#include <list>
+#include <memory>
+#include <rocksdb/iterator.h>
+
+#include "iterlib/Iterator.h"
+
+namespace iterlib {
+
+inline folly::StringPiece sliceToStringPiece(const rocksdb::Slice s) {
+  return {s.data(), s.size()};
+}
+
+class RocksDBIterator : public Iterator {
+ public:
+  RocksDBIterator(rocksdb::Iterator* iter) : iter_(iter) {}
+
+  const Item& key() const override { return keys_.back(); }
```

Comment:
I was suggesting a more deep clone type of method so that we can disconnect the iterator stack from the inner more iterator. 
We can implement this when we need it in a higher level iterator like join or something.